### PR TITLE
Milestone 01: Feature Branch

### PR DESCRIPTION
**Pull request by mavericks-db**
## Feature Branch

This branch includes:
- [x] Follow the tutorial: React Tutorial From Scratch: A Step-by-Step Guide (2021), where you will develop a To-do app!
- [x] Create a new repository and work on a new branch (use Gitflow).

### CODE REVIEW

File: src/classBased/components/TodoContainer.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          return {
+            ...todo,
+            title: updatedTitle,
+          };
+        }
+        return todo;
+      }),
+    }));
+  };
+
```
Comment (Baroka-wp) [inline src/classBased/components/TodoContainer.js:79]: Waou very intelligent. Cheers !

File: src/classBased/components/TodosList.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    PropTypes.shape({
+      id: PropTypes.string,
+      title: PropTypes.string,
+      completed: PropTypes.bool,
+    }),
+  ).isRequired,
+  handleChangeProps: PropTypes.func.isRequired,
+  deleteTodoProps: PropTypes.func.isRequired,
+  setUpdate: PropTypes.func.isRequired,
+};
```
Comment (Baroka-wp) [inline src/classBased/components/TodosList.js:43]: Awesome ! 
Good way to write the propTypes

**Review by Baroka-wp — COMMENTED** [review]
AWESOME ! You are doing great !
Very good job so far... 



Go head and continuous the job ! 
Happy coding, cheers !

**Review by Baroka-wp — APPROVED** [review]
Well Done !


Cheers !

File: src/classBased/components/TodoContainer.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { todos } = this.state;
+    if (prevState.todos !== todos) {
+      const temp = JSON.stringify(todos);
+      localStorage.setItem('todos', temp);
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (mavericks-db) [reply src/classBased/components/TodoContainer.js:33]: Yes, for a functional component. 👍

File: src/classBased/components/TodoContainer.js (hunk 1 above)
Comment (mavericks-db) [reply src/classBased/components/TodoContainer.js:79]: We make use of the spread operator to make an instance of the todos array of objects.

File: src/classBased/components/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    super(props);
+
+    this.state = {
+      editing: false,
+    };
+  }
+
+  componentWillUnmount() {
+    console.log('Cleaning up...');
+  }
```
[the remark this answers is not in the document]
Comment (mavericks-db) [reply src/classBased/components/TodoItem.js:16]: The `console.log` was part of the tutorial code. 👍

**Pull request merged**